src/wolfgame.py

```diff
@@ -1795,108 +1800,108 @@ def fday(cli, nick, chan, rest):
-                        # ends game immediately, with fool as only winner
-                        # hardcode "fool" as the role since game is ending due to them being lynched,
-                        # so we want to show "fool" even if it's a template
-                        lmsg = random.choice(messages["lynch_reveal"]).format(votee, "", "fool")
-                        cli.msg(botconfig.CHANNEL, lmsg)
-                        if chk_win(winner="@" + votee):
-                            return
-                    deadlist.append(votee)
-                    # Other
-                    if votee in get_roles("jester"): # FIXME
-                        var.JESTERS.add(votee)
+            event = Event("chk_decision", {
+                "not_lynching": not_lynching,
+                "votelist": votelist,
+                "numvotes": {}, # filled as part of a priority 1 event
+                "weights": {}, # filled as part of a priority 1 event
+                "transition_night": transition_night
+                }, voters=pl, timeout=False)
+            if not event.dispatch(var, force):
+                return
 
-                    if var.ROLE_REVEAL in ("on", "team"):
-                        rrole = get_reveal_role(users._get(votee)) # FIXME
-                        an = "n" if rrole.startswith(("a", "e", "i", "o", "u")) else ""
-                        lmsg = random.choice(messages["lynch_reveal"]).format(votee, an, rrole)
+            not_lynching = event.data["not_lynching"]
+            votelist = event.data["votelist"]
+            numvotes = event.data["numvotes"]
+
+            gm = var.CURRENT_GAMEMODE.name
+            if (gm == "default" or gm == "villagergame") and len(var.ALL_PLAYERS) <= 9 and var.VILLAGERGAME_CHANCE > 0:
+                if users.Bot in votelist:
+                    if len(votelist[users.Bot]) == avail:
+                        if gm == "default":
+                            channels.Main.send(messages["villagergame_nope"])
+                            stop_game("wolves")
+                            return
+                        else:
+                            channels.Main.send(messages["villagergame_win"])
+                            stop_game("everyone")
+                            return
                     else:
-                        lmsg = random.choice(messages["lynch_no_reveal"]).format(votee)
-                    cli.msg(botconfig.CHANNEL, lmsg)
-                    better_deadlist = [users._get(p) for p in deadlist] # FIXME -- convert chk_decision_lynch to be user-aware
-                    if not del_player(users._get(votee), killer_role="villager", deadlist=better_deadlist, end_game=end_game): # FIXME
-                        return
-                do_night_transision = True
-                break
-        if do_night_transision:
-            event.data["transition_night"](cli)
+                        del votelist[users.Bot]
```

Comment:
The above block could probably be made a high-priority event listener in gamemodes.py (like priority 20) instead of being in wolfgame.py since it's gamemode-specific code. Listener would need to be added to both the default mode as well as the villagergame mode.